custom_components/ble_monitor/ble_parser/airmentor.py

```diff
@@ -11,55 +11,117 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def parse_airmentor(self, data, source_mac, rssi):
+def aqi_to_air_quality(aqi):
+    if aqi <= 50:
+        return "good"
+    elif aqi <= 100:
+        return "moderate"
+    elif aqi <= 150:
+        return "unhealthy for sensitive people"
+    elif aqi <= 200:
+        return "unhealthy"
+    elif aqi <= 250:
+        return "very unhealthy"
+    else:
+        return "hazardous"
+
+
+def parse_pro2(msg_type, xvalue):
+    if msg_type in [0x12, 0x22]:
+        (tvoc, temp, temp_cal, humi, aqi) = unpack(">HHBBH", xvalue)
+        temperature = (temp - 4000) * 0.01
+        temperature_calibrated = temperature - temp_cal * 0.1
+        humi = round(
+            humi * math.exp(temperature * 17.62 / (temperature + 243.12)) / math.exp(
+                temperature_calibrated * 17.62 / (temperature_calibrated + 243.12)
+            ), 2
+        )
+        air_quality = aqi_to_air_quality(aqi)
+        return {
+            # REVIEW NEED: TVOC is ppb in Air Mentor Pro 2, but default using mg/m^3 in Home Assistant.
+            "tvoc": tvoc,
+            "temperature": round(temperature, 2),
+            "temperature calibrated": round(temperature_calibrated, 2),
+            "humidity": round(humi, 2),
+            "aqi": aqi,
+            "air quality": air_quality
+        }
+    elif msg_type in [0x11, 0x21]:
+        (co2, pm25, pm10, co, o3) = unpack(">HHHBB", xvalue)
+        # CO and O3 are not used
+        return {
+            "co2": co2,
+            "pm2.5": pm25,
+            "pm10": pm10,
+        }
+    else:
+        return None
+
+
+def parse_s2(msg_type, xvalue):
+    if msg_type in [0x12, 0x22]:
+        # 2 unknown bytes at the end.
+        (tvoc_ppb, temp, temp_cal, humi, aqi, _, _) = unpack(">HHBBHHH", xvalue)
+
+        # Convert TVOC from ppb to mg/m^3.
+        # ref: https://www.catsensors.com/media/pdf/Sensor_Sensirion_IAM.pdf
+        M_gas = 110
+        V_m = 0.0244 * 1000
+        tvoc_ugm3 = float(tvoc_ppb) * M_gas / V_m
+        tvoc_mgm3 = tvoc_ugm3 / 1000
```

Comment:
I think tvoc should be returned in ug/m3. In `const.py` we have the unit `CONCENTRATION_MICROGRAMS_PER_CUBIC_METER`, which is ug/m3. (mg/m3 = `CONCENTRATION_MILLIGRAMS_PER_CUBIC_METER`

Reply:
oops. Good catch!